experiments/utils.py

```diff
@@ -410,24 +415,19 @@ def _cancel_enrollment(self, experiment):
         alternative = self._get_enrollment(experiment)
         if alternative:
             self.experiment_counter.remove_participant(experiment, alternative, self._participant_identifier())
-            enrollments = self.session.get('experiments_enrollments', None)
-            del enrollments[experiment.name]
-            self.session['experiments_enrollments'] = enrollments
+            self.redis.hdel(self.enrollments_key, experiment.name)
 
     def _experiment_goal(self, experiment, alternative, goal_name, count):
         if self._is_verified_human():
             self.experiment_counter.increment_goal_count(experiment, alternative, goal_name, self._participant_identifier(), count)
         else:
-            goals = self.session.get('experiments_goals', [])
-            goals.append((experiment.name, alternative, goal_name, count))
-            self.session['experiments_goals'] = goals
+            self.redis.lpush(self.goals_key, pickle.dumps((experiment.name, alternative, goal_name, count)))
             logger.info(json.dumps({'type': 'goal_hit_unconfirmed', 'goal': goal_name, 'goal_count': count, 'experiment': experiment.name, 'alternative': alternative, 'participant': self._participant_identifier()}))
 
     def _set_last_seen(self, experiment, last_seen):
-        enrollments = self.session.get('experiments_enrollments', {})
-        alternative, unused, enrollment_date, _ = _session_enrollment_latest_version(enrollments[experiment.name])
-        enrollments[experiment.name] = (alternative, unused, timestamp_from_datetime(enrollment_date), timestamp_from_datetime(last_seen))
-        self.session['experiments_enrollments'] = enrollments
+        enrollment = pickle.loads(self.redis.hget(self.enrollments_key, experiment.name))
```

Comment:
as above, basically if you can compromise the redis data, this can we escalated into and RCE easily, can we use another serialisation format?